src/Middleware/HttpOverrides/src/ForwardedHeadersOptions.cs

```diff
@@ -7,24 +7,27 @@
 
 namespace Microsoft.AspNetCore.Builder
 {
+    /// <summary>
+    /// Options for <see cref="ForwardedHeadersMiddleware"/>
+    /// </summary>
     public class ForwardedHeadersOptions
     {
         /// <summary>
-        /// Use this header instead of <see cref="ForwardedHeadersDefaults.XForwardedForHeaderName"/>
+        /// Gets or sets the header used to retrieve the originating client IP. Defaults to the value specified by 
+        /// <see cref="ForwardedHeadersDefaults.XForwardedForHeaderName"/>.
         /// </summary>
-        /// <seealso cref="Microsoft.AspNetCore.HttpOverrides.ForwardedHeadersDefaults"/>
         public string ForwardedForHeaderName { get; set; } = ForwardedHeadersDefaults.XForwardedForHeaderName;
 
         /// <summary>
-        /// Use this header instead of <see cref="ForwardedHeadersDefaults.XForwardedHostHeaderName"/>
+        /// Gets or sets the header used to retrieve the original value of the Host header field.
+        /// Defaults to the value specified by <see cref="ForwardedHeadersDefaults.XForwardedHostHeaderName"/>
         /// </summary>
-        /// <seealso cref="ForwardedHeadersDefaults"/>
         public string ForwardedHostHeaderName { get; set; } = ForwardedHeadersDefaults.XForwardedHostHeaderName;
 
         /// <summary>
-        /// Use this header instead of <see cref="ForwardedHeadersDefaults.XForwardedProtoHeaderName"/>
+        /// Gets or sets the header used to retrieve the value for the originating scheme (HTTP/HTTPS).
+        /// Defaults to the value specified by <see cref="ForwardedHeadersDefaults.XForwardedProtoHeaderName"/>
```

Comment:
Why not make the same changes to the rest of this file?

Reply:
Updated the other header names. Did you want more details for everything else too? I feel the others didn't quite need as much elaboration.